<a href="https://colab.research.google.com/github/Kassandracw8/Predicting-Sales/blob/main/Project_1_Part_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [8]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [6]:
filename = '/content/sales_predictions - sales_predictions.csv'
sales_predictions = pd.read_csv('/content/sales_predictions - sales_predictions.csv')

In [7]:
sales_predictions.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

In [9]:
sales_predictions2 = sales_predictions.copy()

In [10]:
sales_predictions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


# **Data Cleaning**

In [11]:
sales_predictions.shape

(8523, 12)



*   There are 8523 rows, 12 columns
  



Checking the data types

In [28]:
 data_types = sales_predictions.dtypes
 data_types

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object

There are 3 different data types object, float64, and int64.

In [35]:
str_cols = data_types[data_types=='object'].index
str_cols

Index(['Item_Identifier', 'Item_Fat_Content', 'Item_Type', 'Outlet_Identifier',
       'Outlet_Size', 'Outlet_Location_Type', 'Outlet_Type'],
      dtype='object')

In [39]:
 for col in str_cols:
   print(f'- {col}:')
   print(sales_predictions[col].value_counts(dropna=False))
   print("\n\n")

- Item_Identifier:
FDH26    5
FDL51    5
FDJ09    5
FDB15    5
FDC17    5
        ..
FDE40    1
FDF50    1
NCA42    1
FDK32    1
FDO03    1
Name: Item_Identifier, Length: 1535, dtype: int64



- Item_Fat_Content:
Low Fat    2774
Regular    1575
LF          177
reg          71
low fat      53
Name: Item_Fat_Content, dtype: int64



- Item_Type:
Fruits and Vegetables    670
Snack Foods              656
Household                498
Frozen Foods             477
Dairy                    380
Canned                   361
Baking Goods             351
Health and Hygiene       277
Soft Drinks              241
Meat                     221
Breads                   132
Hard Drinks              117
Others                    91
Starchy Foods             87
Breakfast                 58
Seafood                   33
Name: Item_Type, dtype: int64



- Outlet_Identifier:
OUT013    932
OUT049    930
OUT046    930
OUT035    930
OUT018    928
Name: Outlet_Identifier, dtype: int64



- Outlet_Size:
Small     

In [16]:
sales_predictions.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
8518    False
8519    False
8520    False
8521    False
8522    False
Length: 8523, dtype: bool

In [17]:
sales_predictions.duplicated().sum()

0

In this data set there are not any duplicates.

In [19]:
sales_predictions.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

In [29]:
sales_predictions.isna().sum() / len(sales_predictions) * 100

Item_Identifier              0.0
Item_Weight                  0.0
Item_Fat_Content             0.0
Item_Visibility              0.0
Item_Type                    0.0
Item_MRP                     0.0
Outlet_Identifier            0.0
Outlet_Establishment_Year    0.0
Outlet_Size                  0.0
Outlet_Location_Type         0.0
Outlet_Type                  0.0
Item_Outlet_Sales            0.0
dtype: float64

In [21]:
sales_predictions.dropna(inplace=True)

In [27]:
# Outlet Size
# missing_outlet_size = sales_predictions.Outlet_Size.isna()
# missing_outlet_size
# missing_item_weight = sales_predictions.Item_Weight.isna()
# outlet_size = sales_predictions['Outlet_Size'].value_counts()
# missing_iteam_weight
# sales_predictions['Item_Weight'].value_counts()



*   Replace the missing values with Zero




In [25]:
sales_predictions['Item_Weight'].fillna('Missing', inplace = True)
sales_predictions['Outlet_Size'].fillna('Missing' , inplace = True)
sales_predictions.isna().sum()

Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

In [26]:
item_mrp_mean = sales_predictions['Item_MRP'].mean()
item_mrp_max = sales_predictions['Item_MRP'].max()
item_mrp_min = sales_predictions['Item_MRP'].min()
item_mrp_mean
item_mrp_max
item_mrp_min

31.49

In [34]:
Item_Weight_mean = sales_predictions['Item_Weight'].mean()
Item_Weight_max = sales_predictions['Item_Weight'].max()
Item_Weight_min = sales_predictions['Item_Weight'].min()
Item_Weight_mean
Item_Weight_max
Item_Weight_min

4.555

In [30]:
sales_predictions.describe().round(2)

Item_Weight  Item_Visibility  Item_MRP  Outlet_Establishment_Year  \
count      4650.00          4650.00   4650.00                    4650.00   
mean         12.90             0.06    141.72                    1999.19   
std           4.67             0.04     62.42                       7.39   
min           4.56             0.00     31.49                    1987.00   
25%           8.77             0.03     94.41                    1997.00   
50%          12.65             0.05    142.98                    1999.00   
75%          17.00             0.09    186.61                    2004.00   
max          21.35             0.19    266.89                    2009.00   

       Item_Outlet_Sales  
count            4650.00  
mean             2272.04  
std              1497.96  
min                69.24  
25%              1125.20  
50%              1939.81  
75%              3111.62  
max             10256.65

In [32]:
stats = sales_predictions.describe()
stats.loc[['mean','min','max']]

Item_Weight  Item_Visibility    Item_MRP  Outlet_Establishment_Year  \
mean    12.898675         0.060700  141.716328                1999.190538   
min      4.555000         0.000000   31.490000                1987.000000   
max     21.350000         0.188323  266.888400                2009.000000   

      Item_Outlet_Sales  
mean        2272.037489  
min           69.243200  
max        10256.649000